Source/Scene/TextureAtlas.js

```diff
@@ -353,6 +352,49 @@ function addImage(textureAtlas, image, index) {
   textureAtlas._guid = createGuid();
 }
 
+function getIndex(atlas, image) {
```

Comment:
As a general rule, stuff like this should be called out in the PR description.  I assume by looking at it we had to add some branch new code in order to provide a path for synchronous operations that would otherwise be async in our new "native" Promise world, but in a PR this big; it would be easy for a reviewer to miss.  Also, if someone needs to use revision history to track where/why this code was added, it won't be obvious to them by reading the PR description (which it should be).

Even just a single bullet like "Added TextureAtlas.addImageSync to xxxx" would be extremely valuable to future devs.

This is _especially_ true given the terseness of many of the commit messages in this PR (or perhaps we are squashing) but either way, a paper trail is important.

Reply:
I planned on squashing commits with a more useful commit message. But understood, I'll call out these changes in the PR explicitly.